tests/TestMembershipContextFactory.php

```diff
@@ -4,59 +4,79 @@
 
 namespace WMDE\Fundraising\MembershipContext\Tests;
 
+use Doctrine\Common\Annotations\AnnotationReader;
 use Doctrine\Common\Annotations\AnnotationRegistry;
 use Doctrine\Common\EventManager;
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\DriverManager;
 use Doctrine\ORM\Configuration;
 use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Tools\Setup;
+use Gedmo\Timestampable\TimestampableListener;
 use WMDE\Fundraising\MembershipContext\Authorization\MembershipTokenGenerator;
 use WMDE\Fundraising\MembershipContext\MembershipContextFactory;
 
 class TestMembershipContextFactory {
 
 	private array $config;
 	private Configuration $doctrineConfig;
-	private Connection $connection;
+	private ?Connection $connection;
 	private MembershipContextFactory $factory;
 	private ?EntityManager $entityManager;
 
-	public function __construct( array $config, Configuration $doctrineConfig ) {
+	public function __construct( array $config ) {
 		$this->config = $config;
-		$this->doctrineConfig = $doctrineConfig;
-
-		$this->connection = DriverManager::getConnection( $this->config['db'] );
-		$this->factory = new MembershipContextFactory( $config, $doctrineConfig );
+		$this->doctrineConfig = Setup::createConfiguration( true );
+		$this->factory = new MembershipContextFactory( $config );
 		$this->entityManager = null;
+		$this->connection = null;
+	}
+
+	public function getConnection(): Connection {
+		if ( $this->connection === null ) {
+			$this->connection = DriverManager::getConnection( $this->config['db'] );
+			$this->factory->registerCustomTypes( $this->connection );
+		}
+		return $this->connection;
 	}
 
 	public function getEntityManager(): EntityManager {
 		if ( $this->entityManager === null ) {
-			AnnotationRegistry::registerLoader( 'class_exists' );
+			$this->entityManager = $this->newEntityManager( $this->factory->newEventSubscribers() );
+		}
+		return $this->entityManager;
+	}
 
-			$this->doctrineConfig->setMetadataDriverImpl( $this->factory->newMappingDriver() );
+	private function newEntityManager( array $eventSubscribers = [] ): EntityManager {
+		AnnotationRegistry::registerLoader( 'class_exists' );
```

Comment:
You can remove this line, it's deprecated and unneeded.

Reply:
assuming you meant to remove the deprecated "AnnotationRegistry::registerLoader( 'class_exists' );",
 not the `setMetadataDriverImpl` call, because that would break things